feat: field autocomplete to mantine

IrakliJani opened the pull request:
Closes: https://github.com/lightdash/lightdash/issues/6741

### Description:

IrakliJani commented:
fix overflows 
![CleanShot 2023-10-05 at 20 38 55@2x](https://github.com/lightdash/lightdash/assets/962095/b329233a-89c2-4a8c-bc31-ba5190bc6ccd)

IrakliJani reviewed (commented):
in drilldown:

![CleanShot 2023-10-10 at 16 54 00@2x](https://github.com/lightdash/lightdash/assets/962095/0f8cf9c1-e0a5-4f63-80b6-28afa930b68f)

in dashboard filters
![CleanShot 2023-10-10 at 16 55 35@2x](https://github.com/lightdash/lightdash/assets/962095/aa8af7e0-94ce-4619-9e9e-4b8cf7e5fec7)

in filters: 
![CleanShot 2023-10-10 at 16 56 01@2x](https://github.com/lightdash/lightdash/assets/962095/b26e8d32-4f6e-49db-895e-1e57233fac1a)

magnew commented on packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/FieldLayoutOptions.tsx:
I saw a minor rendering issue with this one. The tooltip tail is a little messed up. See below. It's not a big deal, but it will come up a lot because of where this lays you. Maybe put the position just `top` or whatever instead of top-start? Or turn off the tail?

<img width="555" alt="Screenshot 2023-10-11 at 14 57 21" src="https://github.com/lightdash/lightdash/assets/1864179/d7a49450-434a-41ca-8743-4185c11f9c91">

IrakliJani replied:
have not found an easy fix for this so I did change to `top` for now

magnew reviewed (commented):
This looks great! Nice work. Much better. I took a close look and left a few suggestions. 

The only thing I would really block on is the name change 😄. As annoying as that is. But if you disagree we can talk.

magnew reviewed (approved):
Approved! My suggestions don't need to block.

IrakliJani commented on packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:
```diff
@@ -280,42 +279,22 @@ const FilterConfiguration: FC<Props> = ({
-                            <FormGroup
-                                style={{ marginBottom: '5px' }}
+                            <ItemSelect
+                                data-testid="field-autocomplete"
```
yep, sounds good!

IrakliJani commented on packages/frontend/src/components/DashboardFilter/FilterConfiguration/TileFilterConfiguration.tsx:
```diff
@@ -230,30 +222,22 @@ const TileFilterConfiguration: FC<Props> = ({
+                                            throw new Error(
+                                                'Expected field to be a Field',
+                                            );
+                                        }
```
I've improved typing to have generic field types that extend Item so this is not going to be necessary anymore. thanks for raising this!

IrakliJani commented on packages/frontend/src/components/common/ItemSelect/index.tsx:
done

magnew replied:
🎉 Thank you!

IrakliJani commented on packages/frontend/src/components/common/ItemSelect/index.tsx:
```diff
@@ -0,0 +1,132 @@
+            icon={item ? <FieldIcon item={item} /> : undefined}
+            placeholder={rest.placeholder ?? 'Search field...'}
+            allowDeselect={false}
+            {...rest}
```
I usually try to put `{...rest}` where overrides makes sense. there is no way to pass `value` or `data` to the `FieldSelect` but overriding `onChange` is not something that we want here

IrakliJani commented:
@magnew to sum up my changes here:

- I renamed component and now it's `FieldSelect` thanks for the suggestion!
- I updated FieldSelect component to be a generic so whatever type you pass to item={dim} you'll get from onChange={(dim) => ...}.
- renamed testid
- fixed tooltip
- fixed tests

magnew commented:
> @magnew to sum up my changes here:
> 
> * I renamed component and now it's `FieldSelect` thanks for the suggestion!
> * I updated FieldSelect component to be a generic so whatever type you pass to item={dim} you'll get from onChange={(dim) => ...}.
> * renamed testid
> * fixed tooltip
> * fixed tests

Excellent. I took a quick look. :shipit:

IrakliJani commented:
tests passing ✅

IrakliJani commented:
doing one more user testing before merging (waiting for render first)

IrakliJani commented:
I found an additional issue with in chart config that compiler was not able to catch which I just fixed.
function had any param and code expected string instead of a field type.

IrakliJani commented:
waiting e2e before merging
